File: shell-examples/src/main/rb/io/vertx/example/shell/starwars/starwars_command.rb
require 'vertx-shell/command_builder'
require 'vertx-shell/shell_service'
require 'vertx-shell/command_registry'

starwars = VertxShell::CommandBuilder.command("starwars").process_handler() { |process|

  # Connect the client
  client = process.vertx().create_net_client()
  client.connect(23, "towel.blinkenlights.nl") { |ar_err,ar|
    if (ar_err == nil)
      socket = ar

      # Ctrl-C closes the socket
      process.interrupt_handler() { |v|
        socket.close()
      }

      #
      socket.handler() { |buff|
        # Push the data to the Shell
        process.write(buff.to_string("UTF-8"))
      }.exception_handler() { |err|
        err.print_stack_trace()
        socket.close()
      }

      # When socket closes, end the command
      socket.close_handler() { |v|
        process.end()
      }
    else
      process.write("Could not connect to remote Starwars server\n").end()
    end
  }
}.build($vertx)

service = VertxShell::ShellService.create($vertx, {
  'telnetOptions' => {
    'host' => "localhost",
    'port' => 3000
  }
})
VertxShell::CommandRegistry.get($vertx).register_command(starwars)
service.start() { |ar_err,ar|
  if (!ar_err == nil)
    ar_err.print_stack_trace()
  end
}

File: shell-examples/src/main/js/io/vertx/example/shell/starwars/starwars_command.js
var CommandBuilder = require("vertx-shell-js/command_builder");
var ShellService = require("vertx-shell-js/shell_service");
var CommandRegistry = require("vertx-shell-js/command_registry");

var starwars = CommandBuilder.command("starwars").processHandler(function (process) {

  // Connect the client
  var client = process.vertx().createNetClient();
  client.connect(23, "towel.blinkenlights.nl", function (ar, ar_err) {
    if (ar_err == null) {
      var socket = ar;

      // Ctrl-C closes the socket
      process.interruptHandler(function (v) {
        socket.close();
      });

      //
      socket.handler(function (buff) {
        // Push the data to the Shell
        process.write(buff.toString("UTF-8"));
      }).exceptionHandler(function (err) {
        err.printStackTrace();
        socket.close();
      });

      // When socket closes, end the command
      socket.closeHandler(function (v) {
        process.end();
      });
    } else {
      process.write("Could not connect to remote Starwars server\n").end();
    }
  });
}).build(vertx);

var service = ShellService.create(vertx, {
  "telnetOptions" : {
    "host" : "localhost",
    "port" : 3000
  }
});
CommandRegistry.get(vertx).registerCommand(starwars);
service.start(function (ar, ar_err) {
  if (!ar_err == null) {
    ar_err.printStackTrace();
  }
});
